FlowCrypt/Controllers/Threads/ThreadDetailsDecorator.swift

```diff
@@ -77,3 +69,46 @@ extension AttachmentNode.Input {
         )
     }
 }
+
+private func makeEncryptionBadge(_ input: ThreadDetailsViewController.Input) -> BadgeNode.Input {
+    let icon: String
+    let text: String
+    let color: UIColor
+    switch input.processedMessage?.messageType {
+    case .error:
+        icon = "lock.open"
+        text = "message_decrypt_error".localized
+        color = .red
+    case .encrypted:
+        icon = "lock"
+        text = "message_encrypted".localized
+        color = .main
+    default:
+        icon = "lock.open"
+        text = "message_not_encrypted".localized
+        color = .warningColor
+    }
+
+    return BadgeNode.Input(
+        icon: icon,
+        text: NSAttributedString.text(from: text, style: .regular(12), color: .white),
+        color: color
+    )
+}
+
+private func makeSignatureBadge(_ input: ThreadDetailsViewController.Input) -> BadgeNode.Input? {
+    input.processedMessage?.signature.map {
```

Comment:
Please avoid `map` for single-value/optional values, I think it would be better dedicated for iterable values for clarity in PR reviews.

Instead better use a guard let with nil return.